File: openapi-generator/scala-gatling/src/gatling/scala/org/openapitools/client/model/FleetVehicleResponse.scala

package org.openapitools.client.model


case class FleetVehicleResponse (
    _externalIds: Option[Map[String, String]],
    /* Harsh event detection setting. */
    _harshAccelSetting: Option[String],
    /* ID of the vehicle. */
    _id: Long,
    /* Name of the vehicle. */
    _name: String,
    _vehicleInfo: Option[FleetVehicleResponseVehicleInfo]
)
object FleetVehicleResponse {
    def toStringBody(var_externalIds: Object, var_harshAccelSetting: Object, var_id: Object, var_name: Object, var_vehicleInfo: Object) =
        s"""
        | {
        | "externalIds":$var_externalIds,"harshAccelSetting":$var_harshAccelSetting,"id":$var_id,"name":$var_name,"vehicleInfo":$var_vehicleInfo
        | }
        """.stripMargin
}

File: openapi-generator/kotlin-spring/src/main/kotlin/org/openapitools/model/FleetVehicleResponse.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import org.openapitools.model.FleetVehicleResponseVehicleInfo
import javax.validation.Valid
import javax.validation.constraints.*

/**
 * A vehicle object as returned for fleet/vehicle
 * @param externalIds 
 * @param harshAccelSetting Harsh event detection setting.
 * @param id ID of the vehicle.
 * @param name Name of the vehicle.
 * @param vehicleInfo 
 */
data class FleetVehicleResponse (

        @get:NotNull 
        @JsonProperty("id") val id: Long,

        @get:NotNull 
        @JsonProperty("name") val name: String,

        @JsonProperty("externalIds") val externalIds: Map<kotlin.String, String>? = null,

        @JsonProperty("harshAccelSetting") val harshAccelSetting: String? = null,

        @JsonProperty("vehicleInfo") val vehicleInfo: FleetVehicleResponseVehicleInfo? = null
) {

}

